### PR TITLE
Use insertIfNotExist of the new interface

### PR DESCRIPTION
**Pull request by MorrisJobke**
Ref #8375

### CODE REVIEW

**Review by rullzer — APPROVED** [review]

File: tests/lib/DB/LegacyDBTest.php [hunk 1]
```diff
@@ -134,140 +134,6 @@ public function testLastInsertId() {
 		// now we can check if the two ids are in correct order
 		$this->assertGreaterThan($id1, $id2);
 	}
-	
-	public function testinsertIfNotExist() {
```
Comment (nickvergessen) [inline tests/lib/DB/LegacyDBTest.php]: are there tests for the connection version of it?

**Review by nickvergessen — APPROVED** [review]

File: tests/lib/DB/LegacyDBTest.php (hunk 1 above)
Comment (MorrisJobke) [reply tests/lib/DB/LegacyDBTest.php]: What do you mean by this?

Comment (nickvergessen) [reply tests/lib/DB/LegacyDBTest.php]: I wondered if `IDBConnection::insertIfNotExist()` has tests

**Comment by nickvergessen** [comment]
Conflicting files
lib/public/DB.php

**Comment by MorrisJobke** [comment]
> Conflicting files
> lib/public/DB.php

Already fixed :P

File: tests/lib/DB/LegacyDBTest.php (hunk 1 above)
Comment (MorrisJobke) [reply tests/lib/DB/LegacyDBTest.php]: I couldn't find one ... let me then migrate those tests over to the IDBConnection.

Comment (MorrisJobke) [reply tests/lib/DB/LegacyDBTest.php]: Migrated them to the new class.

**Review by skjnldsv — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]
:+1:

**Pull request merged**